### PR TITLE
Orbital Onboarding Final (For Review) - Xinghao

### PR DESCRIPTION
**Pull request by Gogo-XD**
Passed Tests:

Driver Tests:
image

Thermal Detection and OS 
image

### CODE REVIEW

File: services/thermal_mgr/thermal_mgr.c [hunk 1]
```diff
@@ -44,18 +45,73 @@ void initThermalSystemManager(lm75bd_config_t *config) {
...
+    return ERR_CODE_QUEUE_FULL;
+  }
+
+  if(event == NULL){
+    return ERR_CODE_INVALID_ARG; 
+  }
+
+  if(thermalMgrQueueHandle == NULL){
+    return ERR_CODE_INVALID_STATE;
+  }
```
Comment (Yarik-Popov) [inline services/thermal_mgr/thermal_mgr.c:54]: These checks should be before the send function call

File: services/thermal_mgr/thermal_mgr.c [hunk 2]
```diff
@@ -44,18 +45,73 @@ void initThermalSystemManager(lm75bd_config_t *config) {
...
+        LOG_IF_ERROR_CODE(readTempLM75BD(config.devAddr, &currTemp));
+        
+        if (errCode == ERR_CODE_SUCCESS){
+          addTemperatureTelemetry(currTemp);
+        }
+      }
+
+      // Interupts
+       else if(event.type == THERMAL_MGR_EVENT_OS_INTERRUPT){ // add to termal_mgr.h file
+        float currTemp = 0.0f;
```
Comment (Yarik-Popov) [inline services/thermal_mgr/thermal_mgr.c:100]: You need to read the temperature here as well

File: services/thermal_mgr/thermal_mgr.c [hunk 3]
```diff
@@ -44,18 +45,73 @@ void initThermalSystemManager(lm75bd_config_t *config) {
...
+        if(currTemp < config.hysteresisThresholdCelsius){
+          safeOperatingConditions();
+        }
+        
+        // Higher temperatures
+        else{
+          overTemperatureDetected();
+        }
+            }
+
```
Comment (Yarik-Popov) [inline services/thermal_mgr/thermal_mgr.c:116]: You will need to log an invalid event here

**Review by Yarik-Popov — CHANGES_REQUESTED** [review]
Overall quite good. Somethings are missing

File: services/thermal_mgr/thermal_mgr.c (hunk 1 above)
Comment (Gogo-XD) [reply services/thermal_mgr/thermal_mgr.c:54]: Fixed

File: services/thermal_mgr/thermal_mgr.c (hunk 2 above)
Comment (Gogo-XD) [reply services/thermal_mgr/thermal_mgr.c:100]: Fixed

File: services/thermal_mgr/thermal_mgr.c (hunk 3 above)
Comment (Gogo-XD) [reply services/thermal_mgr/thermal_mgr.c:116]: Fixed

File: lm75bd/lm75bd.c [hunk 4]
```diff
@@ -27,7 +27,26 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
   /* Implement this driver function */
-
```
Comment (Navtajh04) [inline lm75bd/lm75bd.c:30]: check if temp is NULL

File: lm75bd/lm75bd.c [hunk 5]
```diff
@@ -27,7 +27,26 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
   /* Implement this driver function */
-  
+  error_code_t errCode;
+
+  // Pointer Register
+  uint8_t regAddr = 0x00;
+  RETURN_IF_ERROR_CODE(i2cSendTo(devAddr, &regAddr, 1));
```
Comment (Navtajh04) [inline lm75bd/lm75bd.c]: nit use `sizeof(regAddr)` instead of hardcoding 1

File: lm75bd/lm75bd.c [hunk 6]
```diff
@@ -27,7 +27,26 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
...
+
+  uint8_t buff[2] = {0};
+  RETURN_IF_ERROR_CODE(i2cReceiveFrom(devAddr, buff, 2));
```
Comment (Navtajh04) [inline lm75bd/lm75bd.c]: nit use `sizeof(buff)` instead of hardcoding 2

**Review by Navtajh04 — CHANGES_REQUESTED** [review]

**Review by Yarik-Popov — COMMENTED** [review]
Lgtm, dm me your email and I'll add you the the notion

**Pull request closed without merging**